src/manage.c

```diff
@@ -4465,110 +4516,104 @@ manage_system_report (const char *name, const char *duration,
+                                                    now - cmd_param_1,
+                                                    now,
+                                                    name,
+                                                    report,
+                                                    &slave_error);
         }
       else
         {
           // start and end time
-          return get_osp_performance_string (slave,
-                                             cmd_param_1,
-                                             cmd_param_2,
-                                             name,
-                                             report);
+          return_code = get_osp_performance_string (slave,
+                                                    cmd_param_1,
+                                                    cmd_param_2,
+                                                    name,
+                                                    report,
+                                                    &slave_error);
         }
     }
-
-  /* For simplicity, it's up to the command to do the base64 encoding. */
-  if (params_count == 1)
-    command = g_strdup_printf ("gvmcg %ld %s",
-                               cmd_param_1,
-                               name);
   else
-    command = g_strdup_printf ("gvmcg %ld %ld %s",
-                               cmd_param_1,
-                               cmd_param_2,
-                               name);
-
-  g_debug ("   command: %s", command);
-
-  if ((g_spawn_command_line_sync (command,
-                                  &astdout,
-                                  &astderr,
-                                  &exit_status,
-                                  &err)
-       == FALSE)
-      || (WIFEXITED (exit_status) == 0)
-      || WEXITSTATUS (exit_status))
     {
-      int ret;
-      double load[3];
-      GError *get_error;
-      gchar *output;
-      gsize output_len;
-      GString *buffer;
-
-      g_debug ("%s: gvmcg failed with %d", __func__, exit_status);
-      g_debug ("%s: stdout: %s", __func__, astdout);
-      g_debug ("%s: stderr: %s", __func__, astderr);
-      g_free (astdout);
-      g_free (astderr);
-      g_free (command);
-
-      buffer = g_string_new (FALLBACK_SYSTEM_REPORT_HEADER);
-
-      ret = getloadavg (load, 3);
-      if (ret == 3)
+      if (!g_find_program_in_path ("gvmcg"))
         {
-          g_string_append_printf (buffer,
-                                  "Load average for past minute:     %.1f\n",
-                                  load[0]);
-          g_string_append_printf (buffer,
-                                  "Load average for past 5 minutes:  %.1f\n",
-                                  load[1]);
-          g_string_append_printf (buffer,
-                                  "Load average for past 15 minutes: %.1f\n",
-                                  load[2]);
+          return_code = 7;
```

Comment:
I think it would be simpler to move creating the fallback report here as `get_osp_performance_string` can only return 0 for success or 6 if the connection fails.